x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -821,9 +823,36 @@ protected QueryTranslation asQuery(Range r, boolean onAggs) {
                 if (onAggs) {
                     aggFilter = new AggFilter(at.id().toString(), r.asScript());
                 } else {
+                    Holder<Object> lower = new Holder<>(valueOf(r.lower()));
+                    Holder<Object> upper = new Holder<>(valueOf(r.upper()));
+                    Holder<String> format = new Holder<>(dateFormat(r.value()));
+
+                    // for a date constant comparison, we need to use a format for the date, to make sure that the format is the same
+                    // no matter the timezone provided by the user
+                    if (format.get() == null) {
+                        DateFormatter formatter = null;
+                        if (lower.get() instanceof ZonedDateTime || upper.get() instanceof ZonedDateTime) {
```

Comment:
Is there a chance to have different formats between upper and lower? I think so so it might make sense to have two formatters in case of two different formats.

Reply:
@costing, I tried to test something that has two different formats (like `WHERE date0 > CAST(NOW() - INTERVAL 150 YEARS AS DATE) AND date0 < CAST(NOW() AS DATETIME)`) but couldn't do it. If you have any ideas I can try to break this, please let me know.

Also, we need to have one format only because that's the one to be used in the `range` query. Even if the literals themselves will, in I-don't-know-what-scenario, have two different formats, the `range` query will need only one. In case the `range` query will end up having two different formats for `from` and `to` there will be a `parse_exception` error from ES side. So, in either case, `range` will work with one format only for `format`, `from` and `to` fields (all three need to match).

Reply:
@costin, I tried to test something that has two different formats (like `WHERE date0 > CAST(NOW() - INTERVAL 150 YEARS AS DATE) AND date0 < CAST(NOW() AS DATETIME)`) but couldn't do it. If you have any ideas I can try to break this, please let me know.

Also, we need to have one format only because that's the one to be used in the `range` query. Even if the literals themselves will, in I-don't-know-what-scenario, have two different formats, the `range` query will need only one. In case the `range` query will end up having two different formats for `from` and `to` there will be a `parse_exception` error from ES side. So, in either case, `range` will work with one format only for `format`, `from` and `to` fields (all three need to match).

Reply:
Thanks for explaining.